source/_posts/2017-11-07-which-hass-to-use.markdown

```diff
@@ -0,0 +1,79 @@
+title: "Which installation should I do"
+description: "Should you use hass.io or VirtualEnv or hassbian?"
+date: 2017-11-07 10:00:00 +0000
+date_formatted: "November 07, 2017"
+author: Ryan McLean
+author_twitter: ryanm101
+comments: true
+categories: How-To
+---
+
+# Decisions, Decisions, Decisions...
+
+Ok you've made it this far you're interested in automating your home, but Home Assistant has you stuck there are 3 main ways to install it. Which should you choose?
+
+*   Hass.io
+*   Python Virtual Environment
+*   Hassbian
+
+This brief article will hopefully give you a starting point to make your decision as all 3 are perfectly viable options. These are not prescriptive and hass.io for example can be installed on a generic linux server (subject to some [limitations](https://github.com/home-assistant/hassio-build/tree/master/install#supported-machine-types))but if you are looking to do that, you don't need this article to tell you the options.
+
+<!--more-->
+
+## I want the quick answer with no Fluff.
+
+Simply choose where you fall into:
+
+*   I don't care I just want an autonomous home using my RPi / NUC -> Hass.io
+*   I don't want to fully dedicate my Pi and I want a simple install -> Hassbian
+*   I'm a developer or Poweruser / Tinkerer -> Python Virtual Env
+
+## Considerations
+
+The primary consideration in my opinion is *how* do you want to use Home Assistant?
+
+In my case I want a simple stable install for my "production" instance and I have a few spare Raspberry Pis, So I use Hass.io for that "production" instance.
+
+On my Linux Laptop I use the Python Virtual Environment option so I can test different configs and I also have multiple Virtual Environments on the laptop allowing me to develop components using different versions of the libraries.
+
+## Install Options
+### Option 1: [Hass.io](https://home-assistant.io/hassio/)
+
+This is the simplest option, it provides a simplified interface allowing easy installation of addons and the only "Code" you have to worry about is the configuration files. Even updating is just just a simple click in the UI.
+
+This is the best option for beginners or people that want that appliance "just works feel".
+
+For those that like to get into the nitty gritty and tinker, it's the worst option as Hass.io is fully self contained so you lose access to the underlying Linux system. It may also have issues with certain USB devices.
+
+### Option 2: [Hassbian](https://home-assistant.io/docs/hassbian/installation/)
+
+You have a raspberry Pi, you want to be able to see and edit the internals but you want a simple install and you want to use your RPi for multiple things.
+
+Simply download the image and install on the SD card, you can now edit the configs and also tweak code etc as needed.
+
+This option is basically Option3 for lazy people :), basically the image has the virtual environment already setup and configured on your behalf so you don't need to worry about the commands to execute it.
```

Comment:
There's quite a bit more. 
 - Since quite a a few components have some external dependencies these are pre-installed in quite a few cases.
 - Home Assistant is run by a separate user that has limited privileges. 
There's more but it should be highlighted over a Venv.